pages/_app.tsx

```diff
@@ -0,0 +1,35 @@
+import { ChainId, ThirdwebProvider } from "@thirdweb-dev/react";
+import { MagicConnector } from "@thirdweb-dev/react/evm/connectors/magic";
+import type { AppProps } from "next/app";
+import "../styles/globals.css";
+
+// This is the chainId your dApp will work on.
+const activeChainId = ChainId.Mumbai;
+
+const magicLinkConnector = new MagicConnector({
+  options: {
+    apiKey: process.env.NEXT_PUBLIC_MAGIC_LINK_API_KEY as string,
+    rpcUrls: {
+      [ChainId.Mumbai]: "https://rpc-mumbai.maticvigil.com",
+    },
+  },
+});
+
+// Array of wallet connectors you want to use for your dApp.
+const connectors = [magicLinkConnector];
+
+function MyApp({ Component, pageProps }: AppProps) {
+  return (
+    <ThirdwebProvider
+      desiredChainId={activeChainId}
+      walletConnectors={connectors}
+      chainRpc={{
+        [ChainId.Mumbai]: "https://rpc-mumbai.maticvigil.com",
```

Comment:
Any reason we're manually overwriting the RPC URL for Mumbai here? Otherwise it would default to our thirdweb rpc URL.